src/model/ModelObject.cpp

```diff
@@ -851,6 +896,30 @@ namespace model {
     return getImpl<detail::ModelObject_Impl>()->removeAdditionalProperties();
   }
 
+  boost::optional<std::string> ModelObject::cadObjectId() const {
+    return getImpl<detail::ModelObject_Impl>()->cadObjectId();
+  }
+
+  bool ModelObject::setCADObjectId(std::string cadObjectId) {
+    return getImpl<detail::ModelObject_Impl>()->setCADObjectId(cadObjectId);
```

Comment:
Same for public side, use a const ref, (or take by val but `std::move` it)
```suggestion
  bool ModelObject::setCADObjectId(const std::string& cadObjectId) {
    return getImpl<detail::ModelObject_Impl>()->setCADObjectId(cadObjectId);
```